### PR TITLE
Fix parsing error when periodic=False and boxsize not given in the theory module

### PR DESCRIPTION
**Pull request by lgarrison**
From #256, don't pass a boxsize of `None` to the C extensions.

### CODE REVIEW

File: Corrfunc/theory/vpf.py [hunk 1]
```diff
@@ -200,8 +200,12 @@ def vpf(rmax, nbins, nspheres, numpN, seed,
         msg = "Number of counts-in-cells wanted must be at least 1"
         raise ValueError(msg)
 
-    if periodic and boxsize is None:
-        raise ValueError("Must specify a boxsize if periodic=True")
```
Comment (manodeep) [inline Corrfunc/theory/vpf.py:204]: Writing out this way seems easier to understand - any reason for moving this check into the kwargs section?

**Review by manodeep — COMMENTED** [review]
Thanks for the quick fix - everything looks good except for that one change in `vpf.py`. Might be better to keep the ``periodic and boxsize=None`` check separate from the `kwargs` - that will be easier to parse for the reader and be consistent with the remainder of the counters

File: Corrfunc/theory/vpf.py (hunk 1 above)
Comment (lgarrison) [reply Corrfunc/theory/vpf.py:204]: It was just a style preference; having two adjacent conditionals that share a boolean to me says they could/should be merged! But I don't disagree that this way is more explicit.

**Comment by manodeep** [comment]
Unsure why the astropy bots are complaining - the milestone is set and a changelog entry is present.

**Review by manodeep — APPROVED** [review]

**Comment by manodeep** [comment]
@lgarrison This is ready to merge right?

**Comment by lgarrison** [comment]
Yes!

**Pull request merged**